### ISSUE TITLE
FaIR1 adapter: problem with unit conversion

### ISSUE THREAD

**Issue by rgieseke** [openscm/openscm-runner#32]
When running the FaIR1 adapter with a scenario that has e.g. 'Emissions|CO2' in addition to 'Emissions|CO2|MAGICC AFOLU' i think there is a problem as it tries to convert the unit with whatever was the previous unit.

I think it needs to handle the Exception differently, maybe always with `continue` or reset `fair_unit`?
If an Exception occurs, `fair_unit` is compared against the previous value (unless in the list of vars below).

https://github.com/openscm/openscm-runner/blob/master/src/openscm_runner/adapters/fair_adapter/_scmdf_to_emissions.py#L215

```python
for var_df in scmrun.groupby("variable"):
        variable = var_df.get_unique_meta("variable", no_duplicates=True)
        in_unit = var_df.get_unique_meta("unit", no_duplicates=True)
        try:
            fair_col, fair_unit, context = _get_fair_col_unit_context(variable)
        except AssertionError:
            if variable.endswith(
                (
                    "|HFC152a",
                    "|HFC236fa",
                    "|HFC365mfc",
                    "|NF3",
                    "|C3F8",
                    "|C4F10",
                    "|C5F12",
                    "|C7F16",
                    "|C8F18",
                    "|cC4F8",
                    "|SO2F2",
                    "|CH2Cl2",
                    "|CHCl3",
                )
            ):
                continue
    
    if in_unit != fair_unit:
        var_df_fair_unit = var_df.convert_unit(fair_unit, context=context)
    else:
        var_df_fair_unit = var_df
```

Thoughts, @chrisroadmap?

**Comment by chrisroadmap** [comment]
Thanks for picking this up @rgieseke - do you have a scenario in mind I can test this with? Is the issue that it tries to run FaIR with both and fails down the line?

**Comment by chrisroadmap** [comment]
Fixed and ready in #33
